### ISSUE TITLE
[Calico/VPP NSM integration] `memif` socket ID generation is wrong for the single VPP scenario

### ISSUE THREAD

**Issue by Bolodya1997** [networkservicemesh/sdk-vpp#361]
## Issue
Currently `memif` generates socket IDs incrementing some per-application counter:
https://github.com/networkservicemesh/sdk-vpp/blob/914e91ee4c30415045ede2c392b6b56039886bde/pkg/networkservice/mechanisms/memif/common.go#L54

It is totally incorrect for the single VPP scenario (e.g. with Calico VPP) - Client, Forwarder and Endpoint start requesting VPP with the same socket IDs and fail.

## Parent issue
https://github.com/networkservicemesh/integration-k8s-kind/issues/325

## Possible solution
Use some random generator for creating socket IDs + check for "already exists" error to retry:
```go
var random *rand.Rand

func init() {
	random = rand.New(rand.NewSource(time.Now().UnixNano()))
}

func createMemifSocket(ctx context.Context, mechanism *memifMech.Mechanism, vppConn api.Connection, isClient bool) (socketID uint32, err error) {
	// Extract the socket filename
	u, err := url.Parse(mechanism.GetSocketFileURL())
	if err != nil {
		return 0, errors.Wrapf(err, "not a valid url %q", mechanism.GetSocketFileURL())
	}
	if u.Scheme != memifMech.SocketFileScheme {
		return 0, errors.Errorf("socket file url must have scheme %q, actual %q", memifMech.SocketFileScheme, u.Scheme)
	}

	now := time.Now()
	for {
		// Create the socketID
		socketID = random.Uint32()
		memifSocketAddDel := &memif.MemifSocketFilenameAddDel{
			IsAdd:          true,
			SocketID:       socketID,
			SocketFilename: u.Path,
		}

		if _, err = memif.NewServiceClient(vppConn).MemifSocketFilenameAddDel(ctx, memifSocketAddDel); err != nil {
			if err.(api.VPPApiError) == api.ENTRY_ALREADY_EXISTS {
				continue
			}
			return 0, errors.WithStack(err)
		}

		log.FromContext(ctx).
			WithField("SocketID", memifSocketAddDel.SocketID).
			WithField("SocketFilename", memifSocketAddDel.SocketFilename).
			WithField("IsAdd", memifSocketAddDel.IsAdd).
			WithField("duration", time.Since(now)).
			WithField("vppapi", "MemifSocketFilenameAddDel").Debug("completed")
		store(ctx, isClient, memifSocketAddDel)

		return socketID, nil
	}
}
```

**Comment by Bolodya1997** [comment]
@denis-tingaikin @edwarnicke 
cc

**Comment by Bolodya1997** [comment]
Possible solution with generating socket ID on the VPP side - https://gerrit.fd.io/r/c/vpp/+/32271/6/src/plugins/memif/memif.api#43.

**Comment by Bolodya1997** [comment]
Should be closed with #370

**Comment by denis-tingaikin** [comment]
It seems like all PRs have merged.
